library/alloc/src/collections/btree/search.rs

```diff
@@ -94,7 +94,7 @@ impl<BorrowType: marker::BorrowType, K, V> NodeRef<BorrowType, K, V, marker::Lea
         K: Borrow<Q>,
         R: RangeBounds<Q>,
     {
-        // WARNING: Inlining these variables would be unsound (#81138)
+        // It might be unsound to inline these variables if this logic changes (#81138).
```

Comment:
> What logic?

I meant the code in the rest of the method. This might be better:

    // Inlining these variables should be avoided (#81138).

The rest of the comment already explains why inlining can cause a safety issue.